### PR TITLE
Provide web endpoints to view targets discovered using k8s service di…

### PR DESCRIPTION
**Pull request by gawicks**
…scovery #592

image

### CODE REVIEW

File: pkg/server/controller.go [hunk 1]
```diff
@@ -264,11 +273,41 @@ func (ctrl *Controller) serverMux() (http.Handler, error) {
 		{"/metrics", promhttp.Handler().ServeHTTP},
 		{"/exported-metrics", ctrl.exportedMetricsHandler},
 		{"/healthz", ctrl.healthz},
+		{"/targets", ctrl.activeTargetsHandler},
```
Comment (petethepig) [inline pkg/server/controller.go:276]: I think this should be in `diagnosticSecureRoutes`, we don't want everyone to have access to this.

File: webapp/javascript/components/ServiceDiscoveryApp.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  return (
+    <div
+      className="p-2"
+      style={{
+        fontSize: '12px',
+        width: '100%',
+        height: '100%',
+        overflow: 'auto',
+        margin: '0 0 0 20px',
+      }}
```
Comment (petethepig) [inline webapp/javascript/components/ServiceDiscoveryApp.tsx:51]: we're trying to switch to this pattern for styles:
* you put all styles in a module.scss file https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/ExportData.module.scss
* then you import it from your component https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/ExportData.tsx#L11
* then you use classes by names https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/ExportData.tsx#L247

Comment (gawicks) [reply webapp/javascript/components/ServiceDiscoveryApp.tsx:51]: Please ignore the inline styles. I've purposefully left it this way so I could iterate on the design faster!

**Review by eh-am — COMMENTED** [review]
Love this, it's gonna be quite useful!

I have a couple comments:
1. Can you do the data fetching part in a `service`, and do some runtime validations on the data?

Example https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/services/appNames.ts#L22

2. Can you do the redux bit in its own store? Similar to https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/redux/reducers/settings.ts
Also if you could use `redux-toolkit`? Similar to https://github.com/pyroscope-io/pyroscope/blob/2bebc82be08641dd9c4901675555901b5781740b/webapp/javascript/components/NameSelector.tsx#L34

**Review by petethepig — CHANGES_REQUESTED** [review]
Code looks really good. One thing I'm thinking is for consistency's sake I think we should make the output in /targets use `camelCase` (lower case for the first letter).

**Comment by gawicks** [comment]
@petethepig the casing of the output should be fixed in yesterday's commits.

**Comment by dannykopping** [comment]
Sweet, thanks a lot!

**Pull request merged**